platform/ios/src/MGLMapboxEvents.m

```diff
@@ -357,253 +306,198 @@ + (void)pauseMetricsCollection {
     [[MGLMapboxEvents sharedManager] pauseMetricsCollection];
 }
 
-// Must be called from the main thread.
-//
 - (void)pauseMetricsCollection {
-    MGLAssertIsMainThread();
     if (self.paused) {
         return;
     }
+    
     self.paused = YES;
-    [_timer invalidate];
-    _timer = nil;
-    [_eventQueue removeAllObjects];
-    _data = nil;
-    [_session invalidateAndCancel];
-    _session = nil;
+    [self.timer invalidate];
+    self.timer = nil;
+    [self.eventQueue removeAllObjects];
+    self.data = nil;
+    [self.session invalidateAndCancel];
+    self.session = nil;
     
     [self validateUpdatingLocation];
 }
 
 - (void)stopUpdatingLocation {
-    [_locationManager stopUpdatingLocation];
+    [self.locationManager stopUpdatingLocation];
     
     // -[CLLocationManager stopMonitoringVisits] is only available in iOS 8+.
-    if ([_locationManager respondsToSelector:@selector(stopMonitoringVisits)]) {
-        [_locationManager stopMonitoringVisits];
+    if ([self.locationManager respondsToSelector:@selector(stopMonitoringVisits)]) {
+        [self.locationManager stopMonitoringVisits];
     }
     
-    _locationManager = nil;
+    self.locationManager = nil;
 }
 
 + (void)resumeMetricsCollection {
     [[MGLMapboxEvents sharedManager] resumeMetricsCollection];
 }
 
-// Must be called from the main thread.
-//
 - (void)resumeMetricsCollection {
-    MGLAssertIsMainThread();
     if (!self.paused || ![[self class] isEnabled]) {
         return;
     }
+    
     self.paused = NO;
-    _data = [[MGLMapboxEventsData alloc] init];
-    _session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:self delegateQueue:nil];
+    self.data = [[MGLMapboxEventsData alloc] init];
+    self.session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:self delegateQueue:nil];
     
     [self validateUpdatingLocation];
 }
 
 - (void)startUpdatingLocation {
-    MGLAssertIsMainThread();
-    if (_locationManager || _paused) {
-        NSAssert(!(_locationManager && _paused),
+    if (self.locationManager || self.paused) {
+        NSAssert(!(self.locationManager && self.paused),
                  @"MGLMapboxEvents should not have a CLLocationManager while paused.");
         return;
     }
-    _locationManager = [[CLLocationManager alloc] init];
-    _locationManager.desiredAccuracy = kCLLocationAccuracyKilometer;
-    _locationManager.distanceFilter = 10;
-    _locationManager.delegate = self;
+    self.locationManager = [[CLLocationManager alloc] init];
+    self.locationManager.desiredAccuracy = kCLLocationAccuracyKilometer;
+    self.locationManager.distanceFilter = 10;
+    self.locationManager.delegate = self;
```

Comment:
Nit: Let’s reduce the repeated calls to `-locationManager` with a local variable that gets assigned to the property at the end.